### PR TITLE
Add snapshot test docs / Update beta criteria for data accelerators

### PR DESCRIPTION
**Pull request by phillipleblanc**
## 🗣 Description

Adds docs on how to update the snapshots generated by `insta` for the snapshot tests.

Also proposed updates to the beta criteria for data accelerators.

### CODE REVIEW

File: docs/criteria/accelerator-stages.md [hunk 1]
```diff
@@ -9,51 +9,41 @@ Current supported data accelerators (DuckDB, Postgres, SQlite, Arrow) release st
...
 
 - [ ] Streaming when accelerating/loading
-- [ ] Index
+- [ ] Streaming when querying
+- [ ] Supports primary keys and indexes
+- [ ] Supports federation in a single dataset
+- [ ] Supports push-down federation across multiple datasets within the same accelerator
 - [ ] On conflict behaviors - Drop/Upsert
-- [ ] File mode for SQLite and DuckDB
+- [ ] File mode for supported accelerator engines, i.e. SQLite, DuckDB, etc.
```
Comment (lukekim) [inline docs/criteria/accelerator-stages.md]:
```suggestion
- [ ] File mode for embedded accelerator engines, i.e. SQLite, DuckDB, etc.
```

File: docs/criteria/accelerator-stages.md [hunk 2]
```diff
@@ -9,51 +9,41 @@ Current supported data accelerators (DuckDB, Postgres, SQlite, Arrow) release st
 ### Feature set complete
 
 - [ ] Streaming when accelerating/loading
-- [ ] Index
+- [ ] Streaming when querying
+- [ ] Supports primary keys and indexes
```
Comment (lukekim) [inline docs/criteria/accelerator-stages.md]:
```suggestion
- [ ] Primary keys and indexes
```

File: docs/criteria/accelerator-stages.md [hunk 3]
```diff
@@ -9,51 +9,41 @@ Current supported data accelerators (DuckDB, Postgres, SQlite, Arrow) release st
...
+- [ ] Supports federation in a single dataset
+- [ ] Supports push-down federation across multiple datasets within the same accelerator
```
Comment (lukekim) [inline docs/criteria/accelerator-stages.md]:
```suggestion
- [ ] Single dataset federation
- [ ] Push-down federation across multiple datasets within the same accelerator
```

File: docs/criteria/accelerator-stages.md [hunk 4]
```diff
@@ -9,51 +9,41 @@ Current supported data accelerators (DuckDB, Postgres, SQlite, Arrow) release st
...
 - [ ] Common/core arrow data type supports: Null, all Int types, all Float types, Time32, Time64, Timestamp (with/without TZ), Date32, Date64, Duration, Interval, Binary/LargeBinary/FixedSizeBinary, Utf8/LargeUtf8, List/FixedSizeList/LargeList, Struct, Decimal128/Decimal256
 - [ ] No known major bugs/issues
 
 ### Test Coverage
 
 - [ ] End-to-end test to cover accelerating TPCH-SF1 data from S3 and benchmarking TPCH queries (official and simple).
-  - file mode if applicable
-  - memory mode if applicable
+  - [ ] File mode
+  - [ ] Embedded memory mode
```
Comment (lukekim) [inline docs/criteria/accelerator-stages.md]:
```suggestion
  - [ ] In-Memory mode
```

File: docs/criteria/accelerator-stages.md [hunk 5]
```diff
@@ -9,51 +9,41 @@ Current supported data accelerators (DuckDB, Postgres, SQlite, Arrow) release st
...
-    - [ ] clickbench_1 (from datafusion benchmark)
-    - [ ] Semi-structure/Blob based data with row size 1M, 10M, 100M
-  - Can run smoke test on accelerating
-    - [ ] Load SF10 into memory
-    - [ ] Load SF100 into file
-  - Can run smoke test on querying on accelerated data
-    - [ ] Query on SF10 dataset with memory mode
-    - [ ] Query on SF100 dataset with file mode
+- [ ] A reproducible test script that can load a variety of datasets into accelerators in both file/embedded memory mode.
+  - Loads and runs these benchmarks with no errors:
```
Comment (lukekim) [inline docs/criteria/accelerator-stages.md:29]:
```suggestion
  - Loads and runs benchmarks with no errors:
```

File: docs/criteria/accelerator-stages.md [hunk 6]
```diff
@@ -9,51 +9,41 @@ Current supported data accelerators (DuckDB, Postgres, SQlite, Arrow) release st
...
 - [ ] Integration tests to cover "On Conflict" behaviors.
-- [ ] A reproducible test script that can load a variety of datasets into accelerators in both file/memory mode.
-  - Can load below datasets
-    - [ ] TPCH-SF10
-    - [ ] TPCH-SF100
-    - [ ] TPCDS-SF10
-    - [ ] TPCDS-SF100
-    - [ ] db-benchmark join-datagen 50G (from datafusion benchmark)
-    - [ ] clickbench_1 (from datafusion benchmark)
-    - [ ] Semi-structure/Blob based data with row size 1M, 10M, 100M
```
Comment (lukekim) [inline docs/criteria/accelerator-stages.md:33]: I think we still need blob for content?

File: docs/criteria/accelerator-stages.md [hunk 7]
```diff
@@ -9,51 +9,41 @@ Current supported data accelerators (DuckDB, Postgres, SQlite, Arrow) release st
...
-  - Can run smoke test on querying on accelerated data
-    - [ ] Query on SF10 dataset with memory mode
-    - [ ] Query on SF100 dataset with file mode
+- [ ] A reproducible test script that can load a variety of datasets into accelerators in both file/embedded memory mode.
+  - Loads and runs these benchmarks with no errors:
+    - [ ] TPCH-SF10 (embedded memory/file)
+    - [ ] TPCH-SF100 (file)
+    - [ ] TPCDS-SF10 (embedded memory/file)
+    - [ ] TPCDS-SF100 (file)
+    - [ ] Clickbench
```
Comment (lukekim) [inline docs/criteria/accelerator-stages.md]:
```suggestion
  - Loads and runs benchmarks with no errors:
    - [ ] TPCH-SF10 (embedded memory/file)
    - [ ] TPCH-SF100 (file)
    - [ ] TPCDS-SF10 (embedded memory/file)
    - [ ] TPCDS-SF100 (file)
    - [ ] Clickbench
```

Comment (lukekim) [inline docs/criteria/accelerator-stages.md]: Does Clickbench have a variant and/or SF? Does it need to be file based?

File: docs/criteria/accelerator-stages.md [hunk 8]
```diff
@@ -9,51 +9,41 @@ Current supported data accelerators (DuckDB, Postgres, SQlite, Arrow) release st
...
 - [ ] No known major bugs/issues
 
 ### Test Coverage
 
 - [ ] End-to-end test to cover accelerating TPCH-SF1 data from S3 and benchmarking TPCH queries (official and simple).
-  - file mode if applicable
-  - memory mode if applicable
+  - [ ] File mode
+  - [ ] Embedded memory mode
 - [ ] Integration tests to cover accelerating data from S3 parquet, MySQL, Postgres with arrow types: Null, all Int types, all Float types, Time32, Time64, Timestamp (with/without TZ), Date32, Date64, Duration, Interval, Binary/LargeBinary/FixedSizeBinary, Utf8/LargeUtf8, List/FixedSizeList/LargeList, Struct, Decimal128/Decimal256
```
Comment (lukekim) [inline docs/criteria/accelerator-stages.md]: Should split the data connectors to sub tasks. Is this the source of truth for data types?

File: docs/criteria/accelerator-stages.md [hunk 9]
```diff
@@ -9,51 +9,41 @@ Current supported data accelerators (DuckDB, Postgres, SQlite, Arrow) release st
...
+- [ ] Supports primary keys and indexes
+- [ ] Supports federation in a single dataset
+- [ ] Supports push-down federation across multiple datasets within the same accelerator
 - [ ] On conflict behaviors - Drop/Upsert
-- [ ] File mode for SQLite and DuckDB
+- [ ] File mode for supported accelerator engines, i.e. SQLite, DuckDB, etc.
 - [ ] Common/core arrow data type supports: Null, all Int types, all Float types, Time32, Time64, Timestamp (with/without TZ), Date32, Date64, Duration, Interval, Binary/LargeBinary/FixedSizeBinary, Utf8/LargeUtf8, List/FixedSizeList/LargeList, Struct, Decimal128/Decimal256
 - [ ] No known major bugs/issues
 
 ### Test Coverage
```
Comment (lukekim) [inline docs/criteria/accelerator-stages.md:21]: Should there be a statement regarding the use of indexes in accelerators when running benchmarks?

File: docs/criteria/accelerator-stages.md (hunk 7 above)
Comment (lukekim) [reply docs/criteria/accelerator-stages.md]: We should add `Spicebench` for the specific edge cases we need to support, including DECIMAL256, etc.

Comment (phillipleblanc) [reply docs/criteria/accelerator-stages.md]: No, there is only one variant/size (ref: https://github.com/ClickHouse/ClickBench#limitations). I'll clarify this should be run in both memory/file mode.

File: docs/criteria/accelerator-stages.md (hunk 8 above)
Comment (phillipleblanc) [reply docs/criteria/accelerator-stages.md]: > Should split the data connectors to sub tasks

👍

> Is this the source of truth for data types?

Its the list of types we decided to support, so its the source of truth for that. There are other Arrow types/esoteric DB types that we don't claim to support (i.e. Postgres geometry types, Arrow StringView, etc)

**Review by lukekim — APPROVED** [review]

**Pull request merged**